### PR TITLE
docs update ceph-deploy reference to reflect ceph-volume API

### PR DESCRIPTION
**Pull request by alfredodeza**
ceph-deploy now uses ceph-volume as a backend, and a bunch of things are no longer supported like deploying to a directory.

The API has been changed heavily so this helps clarify the new way of doing this in Luminous.

Fixes https://tracker.ceph.com/issues/16466

### CODE REVIEW

**Comment by alfredodeza** [comment]
jenkins render docs

File: doc/man/8/ceph-deploy.rst [hunk 1]
```diff
@@ -251,93 +247,48 @@ Subcommand ``list`` lists disk partitions and Ceph OSDs.
...
 
-Subcommand ``prepare`` prepares a directory, disk or drive for a Ceph OSD. It
-first checks against multiple OSDs getting created and warns about the
-possibility of more than the recommended which would cause issues with max
-allowed PIDs in a system. It then reads the bootstrap-osd key for the cluster or
-writes the bootstrap key if not found. It then uses :program:`ceph-disk`
-utility's ``prepare`` subcommand to prepare the disk, journal and deploy the OSD
-on the desired host. Once prepared, it gives some time to the OSD to settle and
-checks for any possible errors and if found, reports to the user.
+Subcommand ``create`` prepares a device a Ceph OSD. It first checks against
```
Comment (vasukulkarni) [inline doc/man/8/ceph-deploy.rst]: s/device a Ceph OSD/device for Ceph OSD

File: doc/man/8/ceph-deploy.rst [hunk 2]
```diff
@@ -251,93 +247,48 @@ Subcommand ``list`` lists disk partitions and Ceph OSDs.
...
-	ceph-deploy osd prepare HOST:DISK[:JOURNAL] [HOST:DISK[:JOURNAL]...]
+Bluestore Usage::
 
-Subcommand ``activate`` activates the OSD prepared using ``prepare`` subcommand.
-It actually uses :program:`ceph-disk` utility's ``activate`` subcommand with
-appropriate init type based on distro to activate the OSD. Once activated, it
-gives some time to the OSD to start and checks for any possible errors and if
-found, reports to the user. It checks the status of the prepared OSD, checks the
-OSD tree and makes sure the OSDs are up and in.
+	ceph-deploy osd create --data DISK HOST
```
Comment (vasukulkarni) [inline doc/man/8/ceph-deploy.rst]: We could add a line that explains the other 2 optional parameters

"--block-db /path/to/db-device --block-wal /path/to/wal-device"

**Comment by vasukulkarni** [comment]
Overall LGTM

File: doc/man/8/ceph-deploy.rst (hunk 2 above)
Comment (alfredodeza) [reply doc/man/8/ceph-deploy.rst]: I am not adding information on every flag, but I did add that "other flags available, please see the man page or the --help menu on ceph-deploy osd create" which I think should suffice

File: doc/man/8/ceph-deploy.rst (hunk 1 above)
Comment (alfredodeza) [reply doc/man/8/ceph-deploy.rst]: this has been fixed

**Pull request merged**